packages/storage-plus/src/map.rs

```diff
@@ -83,6 +86,41 @@ where
     {
         self.key(k).update(store, action)
     }
+
+    /// If you import the proper Map from the remote contract, this will let you read the data
+    /// from a remote contract in a type-safe way using WasmQuery::RawQuery
+    pub fn query(
```

Comment:
I'm OK with this but, wouldn't be better in terms of encapsulation, specificity and whatnot, to have a derived struct from `Map` for implementing this? Someting like `QueryMap`?

Perhaps better (or at least, simpler): Achieve this with a helper (`query_map`) function? Passing the map, extracting the namespace, building the query, etc. Helper can live in the same module / crate than the struct; `pub(module)` can be used for accessibility if needed.

Reply:
I can make another struct. But curious as to why that would be cleaner.

Maybe we can discuss this (Monday) on a short call and then decide.

Reply:
> I can make another struct. But curious as to why that would be cleaner.
> 
> Maybe we can discuss this (Monday) on a short call and then decide.

Sure. I just feel that adding that to `Map`, which is such a fundamental container, is a little loaded. How many times you'd use a `Map`, vs. this `query` method?

Reply:
I reflected on this a bit more, and I see the option of `query_map(map: &Map<K, T>, k: K) -> StdResult<Option<T>>` as a reasonable alternative, also for item.

however, it seems to only really serve to hide it. One more import and `query_map(&BALANCES, &addr)?` over `BALANCES.query(&addr)?` is a little uglier.

I think the request from Orkun was that these be easy to use, and I would go for adding them here. I will also try to reduce the size of the code, as most of this has nothing to do with map, but overriding `QuerierWrapper.custom_query` to not parse the data (and assume it is non-empty). I will lower the size of code in these methods via a helper, but keep them exposed as methods for usability